upgraded out of date nuget packages

milarso81 opened the pull request:
updating nuget packages

Commit:
upgraded out of date nuget packages

heaths commented on src/VSSetup.PowerShell/VSSetup.PowerShell.csproj:
```diff
@@ -102,6 +102,8 @@
   <ItemGroup>
     <Analyzer Include="..\..\packages\StyleCop.Analyzers.1.0.2\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll" />
     <Analyzer Include="..\..\packages\StyleCop.Analyzers.1.0.2\analyzers\dotnet\cs\StyleCop.Analyzers.dll" />
+    <Analyzer Include="..\..\packages\StyleCop.Analyzers.1.1.118\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll" />
```
These should replace the old ones instead of running both sets of analyzers.

heaths commented on src/VSSetup.PowerShell/VSSetup.PowerShell.csproj:
```diff
@@ -37,7 +37,7 @@
   <ItemGroup>
     <Reference Include="Microsoft.VisualStudio.Setup.Configuration.Interop, Version=1.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Microsoft.VisualStudio.Setup.Configuration.Interop.1.11.2290\lib\net35\Microsoft.VisualStudio.Setup.Configuration.Interop.dll</HintPath>
+      <HintPath>..\..\packages\Microsoft.VisualStudio.Setup.Configuration.Interop.3.0.4492\lib\net35\Microsoft.VisualStudio.Setup.Configuration.Interop.dll</HintPath>
```
Was there a reason to update this? An old version was used for maximum compatibility. If there's nothing new being used, an old version will continue to work just fine.

heaths replied:
Was there a reason to update this? An old version was used for maximum compatibility. If there's nothing new being used, an old version will continue to work just fine.

heaths commented on src/VSSetup.PowerShell/VSSetup.PowerShell.csproj:
```diff
@@ -112,7 +114,8 @@
     <Error Condition="!Exists('..\..\packages\MicroBuild.Core.0.3.0\build\MicroBuild.Core.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\..\packages\MicroBuild.Core.0.3.0\build\MicroBuild.Core.targets'))" />
-    <Error Condition="!Exists('..\..\packages\Nerdbank.GitVersioning.2.3.38\build\Nerdbank.GitVersioning.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\..\packages\Nerdbank.GitVersioning.2.3.38\build\Nerdbank.GitVersioning.targets'))" />
+    <Error Condition="!Exists('..\..\packages\Microsoft.VisualStudio.Setup.Configuration.Interop.3.0.4492\build\Microsoft.VisualStudio.Setup.Configuration.Interop.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\..\packages\Microsoft.VisualStudio.Setup.Configuration.Interop.3.0.4492\build\Microsoft.VisualStudio.Setup.Configuration.Interop.targets'))" />
+    <Error Condition="!Exists('..\..\packages\Nerdbank.GitVersioning.3.4.244\build\Nerdbank.GitVersioning.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\..\packages\Nerdbank.GitVersioning.3.4.244\build\Nerdbank.GitVersioning.targets'))" />
```
Was there an actual change you needed? There were some behavioral changes between v2 and v3. Unless there are changes you need (or security issues, which for this build-time component is unlikely), it's safer to keep the version there.

milarso81 replied:
No I will revert this, thanks for pointing that out.

heaths replied:
I thought you were going to revert this. I wouldn't take the change if not upgrading for good reason.

milarso81 replied:
we are updating 3rd party packages because during our security review the guidance is that any 3rd party open source libraries are to be updated to the current versions to be both a good example, and to ensure that we have latest bug fixes.

milarso81 replied:
I am going to make sure the Setup reference is on the version it was before because that is not a 3rd party package and as you pointed out it is for backwards compatibility.

Commit:
fixed analyzers so one is used
downgraded VS Setup Interop dll to previous version used
Upgraded unit test project

milarso81 commented:
@heaths the goal here is to make sure any third party packages are on the latest possible version to be in compliance of our security policies for things we ship or make available to customers. I am going to take one pass to make sure only third party packages are updated, and the setup packages are not.

heaths commented:
Having the latest package doesn't make it secure, nor would I expect the security policy to state you should just use the latest without regard for any known CVEs.

The concern here is that you are upgrading to a new major version, which for a semver-versioned package (which this is) means breaking changes. Have you verified that output isn't adversely affected? If both the binary output and nupkg are still versioned correctly then it's probably a safe change.

There are also build validations that need to be fixed. Seems to just be having to fully qualify namespaces to satisfy the linter (must be a new linter).

Commit:
reverting changes to test project that where not needed

Commit:
addressing compliance issue by upgrading nuget package

Commit:
fixing package to use the correct latest version

heaths commented:
I'm not sure what's wrong with the build. The `NuGet` package provider is still valid, but the "oneget" mention in the build log is outdated. The `platyPS` package is still available via the NuGet package provider as well. I wonder if it'd work if you just replace these two lines:

https://github.com/microsoft/vssetup.powershell/blob/65a71775b367aedc5eef9bfeb6e07b014325223c/build/build.yml#L26-L27

With just `Install-Module -Name platyPS -SkipPublisherCheck -Force -Confirm:$false`

It could also be that the PowerShell installed on that windows-2016 agent is just too old. I know that OneGet was deprecated a while ago, and that older instance might be all it has. Perhaps changing the pipeline to install a more recent version of PowerShell (Core) would help.

Commit:
trying to switch to pwsh to fix build issues

Commit:
removing minmum version to fix build break

Commit:
adding comments

heaths reviewed (approved):
Assuming this works, LGTM. The pipeline doesn't appear to have kicked off, however.

heaths commented:
/azp run Microsoft.vssetup.powershell

heaths commented:
Pipeline is failing to run. According to [docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/powershell?view=azure-devops) the `pwsh` task is indeed valid (I thought it was, but wasn't 100% sure when I suggested it - might've been mixing it up with GitHub Actions). We'll have to take a look.

heaths commented on build/build.yml:
```diff
@@ -22,12 +22,13 @@ steps:
-    Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
+- pwsh: |
+    Install-PackageProvider -Name NuGet -Force
+# install platyPS to gernate help docs from markup files
```
The pipeline is failing to parse this:

```
/build/build.yml: (Line: 28, Col: 5, Idx: 828) - (Line: 28, Col: 82, Idx: 905): While parsing a block mapping, did not find expected key.
```

Spacing is important in YAML, so this needs to shift over in line with the other commands. This should fix it.
```suggestion
    # install platyPS to gernate help docs from markup files
```

heaths committed:
Fix spacing in build/build.yml

Commit:
try removing installing nuget provider since this is most likely not needed

Commit:
merging

heaths commented on build/build.yml:
```diff
@@ -23,8 +23,6 @@ steps:
 - pwsh: |
-    Install-PackageProvider -Name NuGet -Force
-    # install platyPS to gernate help docs from markup files
     Install-Module -Name platyPS -Repository PSGallery -SkipPublisherCheck -Force
```
If that doesn't work, you also might get rid of the explicit repo:
```suggestion
    Install-Module -Name platyPS -SkipPublisherCheck -Force
```
The default repo ordered list on a clean agent should be secure.

Commit:
fixed build warnings

Commit:
enabling xml docs to fix build warning

Commit:
fixing additional compiler warnings

Commit:
fixing addtional compiler warning when building release

Commit:
updating tag used to pull container image for testing

heaths commented:
/azp run Microsoft.vssetup.powershell

heaths commented:
Saw that server-2016 agents are going out of support, though I think our current agent vs2017-server2016 is custom. Still, we can't pull a server-2019 container image on a server-2016 agent, so the agent probably needs to be upgraded. According to https://docs.microsoft.com/azure/devops/pipelines/agents/hosted#software the `windows-2019` image does have VS2019 which should work to build both vssetup.powershell and vswhere (recommend using the same agents and dockerfiles to easy of troubleshooting and sharing test environments), and the agent OS matters less for this repo anyway (fully-managed).

FWIW, the only issue I had building vswhere on a machine with VS2019 and VS2022Preview installed (never had VS2017, IIRC) was during a DEBUG build, but should be easy to resolve if it happens on agents:

```
cl : Command line error D8016: '/ZI' and '/guard:cf' command-line options are incompatible [C:\Users\heaths\src\vswhere\src\vswhere.lib\vswhere.lib.vcxproj]
```

Commit:
try switching to new build image to resolve issues in pipeline

milarso81 committed:
Merge branch 'develop' into upgrade-nuget-packages
